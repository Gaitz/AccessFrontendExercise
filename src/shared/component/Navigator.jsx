import React, { useState } from "react";
import { Link, useHistory } from "react-router-dom";
import styles from "./Navigator.module.css";

const Navigator = () => {
  const [userName, setUserName] = useState("");
  const history = useHistory();

  return (
    <nav className={styles.navigator}>
      <ul>
        <li>
          <Link to="/">Back to List</Link>
        </li>
        <li>
          <form
            data-testid="searchForm"
            onSubmit={(event) => {
              event.preventDefault();
              history.push(`/${userName}`);
            }}
          >
            <label htmlFor="userName">Search user by account: </label>
            <input
              type="search"
              name="userName"
              id="userName"
              onChange={(e) => {
                setUserName(e.target.value);
              }}
            />
            <button type="submit">go</button>
          </form>
        </li>
      </ul>
    </nav>
  );
};

export default Navigator;
