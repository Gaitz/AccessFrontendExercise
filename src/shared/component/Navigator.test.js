import React from "react";
import Navigator from "./Navigator";
import { getByText, render, fireEvent } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => {
    return {
      push: mockHistoryPush,
    };
  },
}));

describe("test Navigator", () => {
  it("snapshot test", () => {
    const { container } = render(
      <Router>
        <Navigator></Navigator>
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  it("go back link to /", () => {
    const { getByText } = render(
      <Router>
        <Navigator></Navigator>
      </Router>
    );
    expect(getByText("Back to List")).toHaveAttribute("href", "/");
  });

  it("search user with history push", () => {
    const { getByLabelText, getByTestId } = render(
      <Router>
        <Navigator></Navigator>
      </Router>
    );

    const QUERY_TEXT = "gaitz";

    const searchInput = getByLabelText("Search user by account:");

    fireEvent.change(searchInput, {
      target: { value: QUERY_TEXT },
    });

    const searchFrom = getByTestId("searchForm");
    expect(searchFrom).toHaveFormValues({
      userName: QUERY_TEXT,
    });

    fireEvent.submit(searchFrom);

    expect(mockHistoryPush).toHaveBeenCalledWith(`/${QUERY_TEXT}`);
  });
});
