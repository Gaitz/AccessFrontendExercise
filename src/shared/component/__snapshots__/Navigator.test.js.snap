// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Navigator snapshot test 1`] = `
<div>
  <nav
    class="navigator"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Back to List
        </a>
      </li>
      <li>
        <form
          data-testid="searchForm"
        >
          <label
            for="userName"
          >
            Search user by account: 
          </label>
          <input
            id="userName"
            name="userName"
            type="search"
          />
          <button
            type="submit"
          >
            go
          </button>
        </form>
      </li>
    </ul>
  </nav>
</div>
`;
